lib/errors/error-collector.js

```diff
@@ -89,75 +89,62 @@ class ErrorCollector {
+   * @param {string} errorType the type of error: "user", "transactionException", "transaction"
+   * @returns {Array} the iterable property from the transaction based on the error type
    */
-  _processUserErrors(transaction, collectedErrors, expectedErrors) {
-    if (transaction.userErrors.length) {
-      for (let i = 0; i < transaction.userErrors.length; i++) {
-        const exception = transaction.userErrors[i]
-        if (this.collect(transaction, exception)) {
-          ++collectedErrors
-          // if we could collect it, then check if expected
-          if (
-            urltils.isExpectedError(this.config, transaction.statusCode) ||
-            errorHelper.isExpectedException(transaction, exception, this.config, urltils)
-          ) {
-            ++expectedErrors
-          }
-        }
-      }
+  _getIterableProperty(transaction, errorType) {
+    let iterableProperty = null
+    if (errorType === 'user' && transaction.userErrors.length) {
+      iterableProperty = transaction.userErrors
     }
-
-    return [collectedErrors, expectedErrors]
+    if (errorType === 'transactionException') {
+      iterableProperty = transaction.exceptions
+    }
+    return iterableProperty
   }
 
   /**
-   * Helper method for processing exceptions on the transaction (transaction.exceptions array)
+   * Helper method for processing errors that are created with .noticeError(), exceptions
+   * on the transaction (transaction.exceptions array), and inferred errors based on Transaction metadata.
    *
-   * @param {Transaction} transaction the transaction being processed
-   * @param {number} collectedErrors the number of errors successfully .collect()-ed
-   * @param {number} expectedErrors the number of collected errors that were expected
-   * @returns {Array.<number>} the updated [collectedErrors, expectedErrors] numbers post-processing
+   * @param {Transaction} transaction the collected exception's transaction
+   * @param {number} collectedErrors the number of errors we've successfully .collect()-ed
+   * @param {number} expectedErrors the number of errors marked as expected in noticeError
+   * @param {string} errorType the type of error to be processed; "user", "transactionException", "transaction"
+   * @returns {Array.<number>} the updated [collectedErrors, expectedErrors] numbers post processing
    */
-  _processTransactionExceptions(transaction, collectedErrors, expectedErrors) {
-    for (let i = 0; i < transaction.exceptions.length; i++) {
-      const exception = transaction.exceptions[i]
-      if (this.collect(transaction, exception)) {
-        ++collectedErrors
-        // if we could collect it, then check if expected
-        if (
-          urltils.isExpectedError(this.config, transaction.statusCode) ||
-          errorHelper.isExpectedException(transaction, exception, this.config, urltils)
-        ) {
-          ++expectedErrors
+  _processErrors(transaction, collectedErrors, expectedErrors, errorType) {
```

Comment:
Let's add a new sub-test suite to https://github.com/amychisholm03/node-newrelic/blob/error-collector/test/unit/errors/error-collector.test.js to cover the various cases of this function, e.g.:

```js
tap.test('_proccessErrors', t => {
	t.test('no iterableProperty and type is transaction', async t => {
		t.pass()
	})

	t.test('interableProperty is null', async t => {
		t.pass()
	})

	t.end()
})
```